packages/babel-plugin/src/utils/state-manager.js

```diff
@@ -684,17 +684,48 @@ export default class StateManager {
 
       this.injectImportInserted = injectName;
     }
-    for (const [_key, { ltr, rtl }, priority] of styles) {
+
+    for (const [_key, styleObj, priority] of styles) {
+      // build metadataProps as before…
+      const metadataProps: Array<t.ObjectProperty> = [];
+      if ((styleObj as any).constKey != null) {
+        metadataProps.push(
+          t.objectProperty(
+            t.identifier('constKey'),
+            t.stringLiteral((styleObj as any).constKey),
+          ),
+        );
+      }
+      if ((styleObj as any).constVal != null) {
+        const val =
+          typeof (styleObj as any).constVal === 'number'
+            ? t.numericLiteral((styleObj as any).constVal)
+            : t.stringLiteral((styleObj as any).constVal);
+        metadataProps.push(
+          t.objectProperty(t.identifier('constVal'), val),
+        );
+      }
```

Comment:
There should never be a `constKey` without a `constVal` right?

```suggestion
      const constKey: ?string = (styleObj as any).constKey
      const constVal: ?number | string = (styleObj as any).constVal
      if (constKey != null && constVal != null) {
        metadataProps.push(
          t.objectProperty(
            t.identifier('constKey'),
            t.stringLiteral((styleObj as any).constKey),
          ),
        );
        const val =
          typeof (styleObj as any).constVal === 'number'
            ? t.numericLiteral(constVal)
            : t.stringLiteral(constVal);
        metadataProps.push(
          t.objectProperty(t.identifier('constVal'), val),
        );
      }
```

Reply:
But also, `constKey` and `constVal` is all one global key-value map and so it probably makes sense to use the `constKey` as an actual key.